### PR TITLE
Clean up database doc

### PR DESCRIPTION
**Pull request by dstansby**
I tried using the database package recently, and found the docs quite dense to read. Hopefully this improves it a bit.

Obviously writing the guides is quite subjective; I'm happy for anyone else to force push changes to the branch, and for none, some or all of this to be merged eventually.

### CODE REVIEW

**Comment by nabobalis** [comment]
CI fail due to known doctest issues.

File: docs/guide/acquiring_data/database.rst [hunk 1]
```diff
@@ -12,37 +12,31 @@ database is used) or a local or remote database server (see the SQLAlchemy
...
+a URL which describes how to connect to the database. The supported
 format of this URL is described by the documentation of
-:func:`sqlalchemy.create_engine` as follows:
-
-    "The string form of the URL is
-    ``dialect+driver://user:password@host/dbname[?key=value..]``, where
-    dialect is a database name such as ``mysql``, ``oracle``, ``postgresql``,
-    etc., and driver the name of a DBAPI, such as ``psycopg2``,
-    ``pyodbc``, ``cx_oracle``, etc."
+:func:`sqlalchemy.create_engine`.
```
Comment (nabobalis) [inline docs/guide/acquiring_data/database.rst]: Would not knowing how to structure the url string be important? 

Maybe that's only important if you already have some sort of remote server with the database on.

Comment (dstansby) [reply docs/guide/acquiring_data/database.rst]: I figured the description is in `sqlalchemy.create_engine`, so if you want to find out it's fairly easy to click on the link.

Comment (nabobalis) [reply docs/guide/acquiring_data/database.rst]: That's fine by me.

**Pull request merged**